drivers/hmis/app/graphql/types/hmis_schema/has_services.rb

```diff
@@ -33,7 +33,8 @@ def resolve_services(scope = object.services, **args)
       private
 
       def apply_service_arguments(scope, sort_order: :date_provided)
-        scope.sort_by_option(sort_order) if sort_order.present?
+        scope = scope.sort_by_option(sort_order) if sort_order.present?
+        scope.viewable_by(current_user)
```

Comment:
This is where I get a little nervous @eanders so I just wan to point this out.

Right now you can only resolve services on enrollments, like this `enrollment { services { ... } }`, so we don't technically *need* a viewable by filter.

If we were to add the ability to view all services for a particular client (or a project), we need to do a couple things
1) Add in a viewable_by filter
2) Use data loaders to prevent n+1 (probably )

So (2) is fine if we forget to do it right away, but (1) is bad. So that's why I'm adding it preemptively here. If we forgot to do (1), a query like `client { services { ... } }` would return services that are tied to enrollments that the user doesn't have view access to.

I would like if there was a better pattern, but this is where it is currently. I added a tests for viewability when resolving related records on Client and Enrollment as part of this PR.

Reply:
I don't believe there's a situation where you could see the services provided but not know which enrollment they were attached to.  Some HMIS allow for services to be attached directly to clients, but in our experience, that tends to cause issues in data exchange and I don't believe we should support that.

Reply:
Right but I'm talking about the schema design. Currently we only list the events on the Enrollment page, so the frontend queries something like this:
```
enrollment(id: 1) {
   events(limit: 10, offset: 0) {
      ...eventinfo
   }
}
```

But if we were to build a page that lists all recent services for a given Client, we would probably want to add a query like this:


```
client(id: 1) {
   events(limit: 10, offset: 0) {
      ...eventinfo
   }
}
```

For the latter query, we _need_ the `viewable_by` filter on the event resolver. For the first one, we don't need it.